core/src/main/java/it/pagopa/selfcare/mscore/core/UserEventServiceImpl.java

```diff
@@ -79,19 +82,45 @@ public void sendLegalTokenUserNotification(Token token) {
             List<UserToNotify> usersToNotify = toUserToNotify(tokenUser.getUserId(), token.getInstitutionId(), token.getProductId(), Optional.empty(), Optional.of(token.getId()));
             usersToNotify.forEach(user -> {
                 UserNotificationToSend notification = notificationMapper.setNotificationDetailsFromToken(token, user, QueueEvent.ADD);
+                String id = user.getUserId().concat("_"+notification.getInstitutionId()).concat("_"+notification.getProductId()).concat("_"+user.getProductRole());
+                notification.setId(id);
                 try {
                     String msg = mapper.writeValueAsString(notification);
                     sendUserNotification(msg, user.getUserId());
                 } catch (JsonProcessingException e) {
                     log.warn(ERROR_DURING_SEND_DATA_LAKE_NOTIFICATION_FOR_USER, user.getUserId());
                 }
             });
-
         });
     }
 
+    @Override
+    public void sendOnboardedUserNotification(OnboardedUser onboardedUser, String productId) {
+        try {
+            User user = userRegistryConnector.getUserByInternalId(onboardedUser.getId(), false);
+            onboardedUser.getBindings().forEach(userBinding -> {
```

Comment:
O(n^2)?

Reply:
the call to the user registry is needed to retrieve the user's PII except for the fiscal code